### DISCUSSION TITLE
Dev environment setup with vscode

### DISCUSSION THREAD

**Discussion by RJ** [toverainc/willow#83] [Q&A]
Using vscode i'm attaching to the willow docker container with the dev containers extension, but vscode is still not sure where to find headers, or how to build. I have the esp-idf extension, cmake, etc too. but i've not used esp-idf much before, and it seems kinda finicky.

Can you advise, or share any dev environment setup you might have to make it easier to get started with (eg) vscode with a change/build/flash loop? 

What does your dev environment / process look like? 

Thanks

**Comment by hamishcunningham** [comment]
Platformio more reliable than idf extension I think, but you'll need to
create a platformio.ini if it isn't there already

**Comment by kristiankielhofner** [comment]
ESP IDF development environments are fairly challenging. Between the container and ```utils.sh``` we do a lot to wrap around it for Willow. Our README provides basic [instructions](https://github.com/toverainc/willow#development) for development. To VS Code the entire willow directory is just another C/kconfig/cmake/python/shell/etc repo.

I would be amazed if someone got Willow to build with the ESP-IDF extension. If you can get it to work that's great but we will not be supporting it directly as of now.
